v5.4.2 skip user management bug

<!--
- Please follow the issue template below for bug reports.
- If you have a support request rather than a bug, please use [Stack Overflow](http://stackoverflow.com/questions/tagged/jhipster) with the JHipster tag.
- For bug reports it is mandatory to run the command `jhipster info` in your project's root folder, and paste the result here.
- Tickets opened without any of these pieces of information will be **closed** without any explanation.
-->

##### **Overview of the issue**

Upon installing jhipster 5.4.2 (skipping user management):

jhipster --skip-user-management

I get the following error:

ERROR Failed to compile with 1 errors 22:02:12

This relative module was not found:

./user-management in ./src/main/webapp/app/shared/reducers/index.ts ✖ ｢wdm｣: 971 modules
ERROR in ./src/main/webapp/app/shared/reducers/index.ts Module not found: Error: Can't resolve './user-management' in '/Users/spasco/code/web/polarisnousers/src/main/webapp/app/shared/reducers' ℹ ｢wdm｣: Failed to compile.

##### **Motivation for or Use Case**

A JHipster 5.4.2 app does not start with "--skip-user-management" enabled.

##### **Reproduce the error**

1. Create a monolithic app with "jhipster --skip-user-management"
2. Select "No" to jhipster registry
3. (for auth) Select JWT auth
4. (for db) Select MongoDB
5. (for cache) Select "No" to spring cache
6. (for build) Select Maven
7. (for other technologies) Select API first development
8. (for web framework) Select React
9 (for SaSS) Select Yes
10. (for il8) Select No
11. (for testing) Select Cucumber and Gatling
12. (other generators) Select No
13. (after app is constructed) run .mvnw

The error will show after completing the above.

##### **Related issues**

No similar issue reported before

##### **Suggest a Fix**

<!-- If you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit) -->

##### **JHipster Version(s)**

5.4.2

##### **JHipster configuration**

Using JHipster version installed locally in current project's node_modules
Executing jhipster:info
Options: from-cli: true
Welcome to the JHipster Information Sub-Generator

##### **JHipster Version(s)**

```
polaris@0.0.0 /Users/spasco/code/web/polarisnousers
└── generator-jhipster@5.4.2 

```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.techrisk.seceng"
    },
    "jhipsterVersion": "5.4.2",
    "applicationType": "monolith",
    "baseName": "polaris",
    "packageName": "com.techrisk.seceng",
    "packageFolder": "com/techrisk/seceng",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "no",
    "enableHibernateCache": false,
    "websocket": false,
    "databaseType": "mongodb",
    "devDatabaseType": "mongodb",
    "prodDatabaseType": "mongodb",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": true,
    "jwtSecretKey": "bXktc2VjcmV0LXRva2VuLXRvLWNoYW5nZS1pbi1wcm9kdWN0aW9uLWFuZC10by1rZWVwLWluLWEtc2VjdXJlLXBsYWNl",
    "clientFramework": "react",
    "useSass": true,
    "clientPackageManager": "npm",
    "testFrameworks": [
      "gatling",
      "cucumber"
    ],
    "jhiPrefix": "jhi",
    "enableTranslation": false,
    "skipUserManagement": true
  }
}
</pre>
</details>


##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>


##### **Environment and Tools**

java version "1.8.0_181"
Java(TM) SE Runtime Environment (build 1.8.0_181-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.181-b13, mixed mode)

git version 2.17.1 (Apple Git-112)

node: v8.12.0

npm: 6.4.1

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

Not applicable

##### **Browsers and Operating System**

<!-- What OS are you on? is this a problem with all browsers or only IE8? -->

MacOs

-   [ x ] Checking this box is mandatory (this is just to show you read everything)

<!-- Love JHipster? Please consider supporting our collective:
👉  https://opencollective.com/generator-jhipster/donate -->

--- a/generators/client/files-react.js
+++ b/generators/client/files-react.js
@@ -393,7 +393,7 @@ const files = {
             templates: ['spec/app/shared/reducers/locale.spec.ts']
         },
         {
-            condition: generator => generator.skipUserManagement,
+            condition: generator => generator.skipUserManagement && generator.authenticationType === 'oauth2',
             path: TEST_SRC_DIR,
             templates: ['spec/app/shared/reducers/user-management.spec.ts']
         },

--- a/generators/client/templates/react/src/main/webapp/app/shared/reducers/index.ts.ejs
+++ b/generators/client/templates/react/src/main/webapp/app/shared/reducers/index.ts.ejs
@@ -28,7 +28,7 @@ import applicationProfile, { ApplicationProfileState } from './application-profi
 import administration, { AdministrationState } from 'app/modules/administration/administration.reducer';
 <%_ if (!skipUserManagement) { _%>
 import userManagement, { UserManagementState } from 'app/modules/administration/user-management/user-management.reducer';
-<%_ } else { _%>
+<%_ } else if (authenticationType === 'oauth2') { _%>
 import userManagement, { UserManagementState } from './user-management';
 <%_ } _%>
 <%_ if (authenticationType !== 'oauth2') { _%>
@@ -50,7 +50,9 @@ export interface IRootState {
   <%_ } _%>
   readonly applicationProfile: ApplicationProfileState;
   readonly administration: AdministrationState;
+  <%_ if (!skipUserManagement || authenticationType === 'oauth2') { _%>
   readonly userManagement: UserManagementState;
+  <%_ } _%>
   <%_ if (authenticationType !== 'oauth2') { _%>
   readonly register: RegisterState;
   readonly activate: ActivateState;
@@ -72,7 +74,9 @@ const rootReducer = combineReducers<IRootState>({
   <%_ } _%>
   applicationProfile,
   administration,
+  <%_ if (!skipUserManagement || authenticationType === 'oauth2') { _%>
   userManagement,
+  <%_ } _%>
   <%_ if (authenticationType !== 'oauth2') { _%>
   register,
   activate,

